### PR TITLE
Avoid php message "Invalid argument supplied for foreach()" 

### PR DESCRIPTION
**Pull request by DeepDiver1975**
refs #15590 

@nickvergessen @MorrisJobke please review - THX

### CODE REVIEW

**Comment by karlitschek** [comment]
makes sense :+1:

File: tests/lib/group/manager.php [hunk 1]
```diff
@@ -846,4 +846,39 @@ public function testGetUserGroupsWithRemoveUser() {
...
+	public function testGetUserIdGroups() {
+		/**
+		 * @var \PHPUnit_Framework_MockObject_MockObject | \OC_Group_Backend $backend
+		 */
+		$backend = $this->getMock('\OC_Group_Database');
+		$backend->expects($this->any())
+			->method('getUserGroups')
+			->with('user1')
+			->will($this->returnValue(null));
+//		$backend->expects($this->any())
```
Comment (MorrisJobke) [inline tests/lib/group/manager.php]: Can't this be removed?

**Comment by nickvergessen** [comment]
Shouldn't the actual fix be to fix the user backend, which does not return an array (which is what the docs say, as well as all default implementations do):
backend.php
```php
	public function getUserGroups($uid) {
		return array();
	}
```

database.php
```php
	public function getUserGroups( $uid ) {
		// No magic!
		$stmt = OC_DB::prepare( "SELECT `gid` FROM `*PREFIX*group_user` WHERE `uid` = ?" );
		$result = $stmt->execute( array( $uid ));

		$groups = array();
		while( $row = $result->fetchRow()) {
			$groups[] = $row["gid"];
		}

		return $groups;
	}
```

**Comment by DeepDiver1975** [comment]
> Shouldn't the actual fix be to fix the user backend, which does not return an array

We have no control over all backends - this is a safety belt.

File: tests/lib/group/manager.php (hunk 1 above)
Comment (DeepDiver1975) [inline tests/lib/group/manager.php]: kill - commit - push :speak_no_evil:

Comment (MorrisJobke) [inline tests/lib/group/manager.php]: Done

**Comment by icewind1991** [comment]
:+1: makes sense

**Pull request merged**